openlibrary/plugins/upstream/account.py

```diff
@@ -959,3 +943,24 @@ def g(*a, **kw):
         finally:
             web.ctx.headers = []
     return g
+
+
+def process_goodreads_csv(i):
+    import csv
+    csv_payload = i.csv if isinstance(i.csv, str) else i.csv.decode()
```

Comment:
```suggestion
    csv_payload = i.csv if isinstance(i.csv, string_types) else i.csv.decode()
```
Near the top of the file add `from six import string_types` to get access to
https://six.readthedocs.io/#six.string_types

Reply:
Summarizing a discussion we had on Slack, for posterity:  

This won't work because Py2's csv library doesn't support Unicode:  https://stackoverflow.com/questions/1846135

We considered 1) adding a new dependency for Py2 csv Unicode support, 2) removing unicode from the test (since anecdotal testing did not show unicode was actually used when testing csv uploads to a Py2 or Py3 OL from Chrome), or 3) skipping the test when using Py2.  We opted for (3).

Reply:
As mentioned above, I split up the test into two (and added a test class).  I only skipped the utf8 test for Python versions < 3.0.  All the tests pass now.  PTAL.